python/tests/integration/test_server.py

```diff
@@ -25,46 +25,61 @@
 import tornado.ioloop
 
 import tchannel.tornado.tchannel as tornado_tchannel
-from tchannel.tornado.broker import ArgSchemeBroker
-from tchannel.tornado.dispatch import RequestDispatcher
 
 
 class Expectation(object):
-    """Represents an expectation for the ServerManager."""
+    """Represents an expectation for the TestServer."""
+
     def __init__(self):
-        # raw message to respond with
-        self.response = None
+        self.execute = None
+
+    def and_write(self, body, headers=None):
+
+        def execute(request, response):
+            if headers:
+                response.write_header(headers)
+
+            response.write_body(body)
+
+        self.execute = execute
+
+    def and_result(self, result):
+
+        def execute(request, response):
+            response.write_result(result)
 
-    def and_return(self, resp):
-        """Write the given Message as a response."""
-        self.response = resp
+        self.execute = execute
 
+    def and_raise(self, exc):
 
-class ServerManager(object):
+        def execute(request, response):
+            raise exc
+
+        self.execute = execute
+
+
+class TestServer(object):
```

Comment:
:clap: and the engineer thus managed to achieve code reuse without inheritance :clap: